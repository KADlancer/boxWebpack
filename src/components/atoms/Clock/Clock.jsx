import * as React from 'react'

// import './Clock.scss'

const STATUS = {
    HOVERED: 'Clock hovered',
    NORMAL: 'Clock normal',
}

class Clock extends React.Component {
    constructor(props) {
        super(props)

        this.onMouseEnter = this.onMouseEnter.bind(this)
        this.onMouseLeave = this.onMouseLeave.bind(this)

        this.state = {
            class: STATUS.NORMAL,
            time: new Date().toLocaleTimeString(),
        }
    }

    componentDidMount() {
        this.intervalID = setInterval(() => this.tick(), 1000)
    }

    componentWillUnmount() {
        clearInterval(this.intervalID)
    }

    onMouseEnter() {
        this.setState({ class: STATUS.HOVERED })
    }

    onMouseLeave() {
        this.setState({ class: STATUS.NORMAL })
    }

    tick() {
        this.setState({
            time: new Date().toLocaleTimeString(),
        })
    }

    render() {
        const { time } = this.state
        return (
            <p
                className={this.state.class}
                onMouseEnter={this.onMouseEnter}
                onMouseLeave={this.onMouseLeave}
            >
                {this.props.children}
                {time}
            </p>
        )
    }
}
export default Clock
