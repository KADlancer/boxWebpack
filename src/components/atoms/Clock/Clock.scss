.Clock {
    margin: 10px;
    padding: 10px;
    border: 1px solid black;
    background: white;
    display: inline-block;

    &.normal {
        background: white;
        color: black;
    }
    &.hovered {
        background: #ff7100;
        color: white;
    }
}
