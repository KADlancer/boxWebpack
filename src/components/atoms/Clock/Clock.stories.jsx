import React from 'react'
import { storiesOf } from '@storybook/react'

import { withTests } from '@storybook/addon-jest';
import results from '../../../../.jest-test-results.json';




import Clock from './index'

storiesOf('Atoms/Clock', module)
    //.addDecorator(withTests({results,}))
    .add('default', () => <Clock>Storytime: </Clock>)
