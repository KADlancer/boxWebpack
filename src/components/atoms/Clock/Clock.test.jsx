import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson } from 'enzyme-to-json'
import { advanceBy, advanceTo, clear } from 'jest-date-mock'
import Clock from './index'

Enzyme.configure({ adapter: new Adapter() })

beforeEach(() => {
    advanceTo(new Date(2018, 5, 27, 0, 0, 0)); // reset to date time.
});
afterEach(() => {
    clear();
});

describe('Link changes the class when hovered', () => {
    it('should render correctly', () => {
        const component = shallow(<Clock>It is: </Clock>)
        const tree = shallowToJson(component)
        expect(tree).toMatchSnapshot()
    })

    it('should set the state to class: STATUS:HOVER if onMouseEnter is called', () => {
        const component = shallow(<Clock>It is: </Clock>)
        component.instance().onMouseEnter()
        expect(component.state('class')).toBe('Clock hovered')
    })
    /*
    // manually trigger the callback

    expect(tree).toMatchSnapshot()

    // manually trigger the callback
    tree.props.onMouseLeave()
    // re-rendering
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    */
})
