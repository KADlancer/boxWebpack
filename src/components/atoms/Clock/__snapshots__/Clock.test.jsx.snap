// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered should render correctly 1`] = `
<p
  className="Clock normal"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  It is: 
  12:00:00 AM
</p>
`;
